go/mcap/writer.go

```diff
@@ -463,7 +463,8 @@ func (w *Writer) flushActiveChunk() error {
 	messageIndexOffsets := make(map[uint16]uint64)
 	if !w.opts.SkipMessageIndexing {
 		for _, chanID := range w.channelIDs {
-			if messageIndex, ok := w.messageIndexes[chanID]; ok {
+			messageIndex, ok := w.messageIndexes[chanID]
+			if ok && !messageIndex.IsEmpty() {
```

Comment:
this seems ok to me, I agree current behavior is likely oversight.

One thing that is unclear to me is, how should we behave if a channel record shows up in a chunk, with no messages?


Another side-note in case it's helpful - I think the extra tooling around messageIndex (Add, IsEmpty, Entries, Reset) can probably be refactored out in favor of a slice of MessageIndexEntry maintained in the writer for each chunk. The extra structure was originally required when MessageIndex was maintaining a sort on the records, but it no longer does that. No need to act on this - just possibly useful context.

Reply:
> how should we behave if a channel record shows up in a chunk, with no messages?

I don't feel like this is unclear to me - the message index does not index channel records, so the presence of channel records shouldn't affect the content or presence of message index records.